### PR TITLE
Fixes #459 : Transistor Amplifier Experiment

### LINKED ISSUE #459: Implement Transistor Amplifier Experiment
**Actual Behaviour**

Transistor Amplifier experiment has not been implemented.

**Expected Behaviour**

Experiment needs to be implemented in the saved experiments

**Steps to reproduce it**

1. Open app
2. Go to `Saved Experiments`
3. Observe `Transistor Amplifier` experiment

**LogCat for the issue**

> Not available

**Screenshots of the issue**

> Not available

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by CloudyPadmal**
Fixes issue #459 

Changes: 
- Implemented similar experiment to the desktop application
- Used functions in oscilloscope activity to capture trace

Screenshots for the change: 
device-2017-08-09-153935

APK for testing:
[app-debug.apk.zip](https://github.com/fossasia/pslab-android/files/1210702/app-debug.apk.zip)

There is a huge distortion happening and can you guys review this PR to see if I have made any errors in implementing the functionalities? Thanks a lot in advance. I have used W1 and W2 to generate waveforms at 200 Hz and 400 Hz respectively. W1 connected to CH2 and W2 connected to CH1.

### CODE REVIEW

**Comment by akarshan96** [comment]
@CloudyPadmal yes, there is a bug causing distorted signals, it's stated under #422. I am trying to resolve this issue. I think the sine wave will be plotted the issue is resolved, there nothing wrong from your end.

**Comment by CloudyPadmal** [comment]
Thank you @akarshan96 for the review and info! :+1:

File: app/src/main/java/org/fossasia/pslab/experimentsetup/TransistorAmplifierExperiment.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+
+        View view = inflater.inflate(R.layout.transistor_amplifier_view, container, false);
+        outputChart = (LineChart) view.findViewById(R.id.line_chart);
+        Button btnConfigure = (Button) view.findViewById(R.id.btn_configure_dialog);
+        btnConfigure.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                MaterialDialog dialog = new MaterialDialog.Builder(getActivity())
+                        .title("Configure Experiment")
```
Comment (viveksb007) [inline app/src/main/java/org/fossasia/pslab/experimentsetup/TransistorAmplifierExperiment.java]: Use string resources

File: app/src/main/java/org/fossasia/pslab/experimentsetup/TransistorAmplifierExperiment.java [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+                        .customView(R.layout.transistor_amplifier_characteristics_dialog, true)
+                        .positiveText("Start Experiment")
```
Comment (viveksb007) [inline app/src/main/java/org/fossasia/pslab/experimentsetup/TransistorAmplifierExperiment.java]: here too.

File: app/src/main/java/org/fossasia/pslab/experimentsetup/TransistorAmplifierExperiment.java (hunk 1 above)
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/experimentsetup/TransistorAmplifierExperiment.java]: Thanks for review @viveksb007 ! I made the requested changes! :)

**Pull request merged**